### ISSUE TITLE
Layout/SpaceInsideReferenceBrackets has false negative, when references nested.

### ISSUE THREAD

**Issue by takumiabe** [bbatsov/rubocop#5914]
`Layout/SpaceInsideReferenceBrackets` doesn't detect the offence, with nested  blackets accessor.

## Expected behavior

It's offence.

## Actual behavior

Offence not detected.

## Steps to reproduce the problem

test.yml
```yaml
Layout/SpaceInsideReferenceBrackets:
  Enabled: true
  EnforcedStyle: no_space
```

test.rb
```rb
record[ options[:attribute] ]
```

and 
```
$ rubocop -c test.yml test.rb --only Layout/SpaceInsideReferenceBrackets
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

```
$ bin/rubocop -V
0.56.0 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by koic** [comment]
Thanks for the feedback and concrete step to reproduction. I confirmed this reproduction and opened a PR #5923. Thank you!

**Comment by takumiabe** [comment]
@koic 
Thank you quick response and fix!
I'm happy :)
